pkg/utils/utils.go

```diff
@@ -251,3 +252,49 @@ func setMap(field reflect.Value, value interface{}) error {
 	field.Set(resultMap)
 	return nil
 }
+
+func TranformGroupName(input string) (string, error) {
+	patterns := []struct {
+		pattern *regexp.Regexp
+		format  func([]string) string
+	}{
+		{
+			regexp.MustCompile(`^dataverse-platform-(.*)$`),
+			func(groups []string) string { return groups[1] + "_group" },
+		},
+		{
+			regexp.MustCompile(`^dataverse-source-(.*)$`),
+			func(groups []string) string { return groups[1] + "_group" },
+		},
```

Comment:
this should be taken from appconfig so that without any code changes, we should be able to configure this.